### DISCUSSION TITLE
Support for turn-by-turn instructions over bluetooth

### DISCUSSION THREAD

**Discussion by Epy** [rinigus/pure-maps#659] [Ideas]
Hello
I didn't find any issue or discussion with this subject I wanted to talk about:

Would it be possible to support sending navigation instructions over bluetooth ? I'm sure it would be a lot of work, but just to know if that's something you would try. Maybe a bounty with other interested users would be possible and help to develop this ?
As more and more vehicles are compatible, it seems a good way to use OSM maps and this wonderful software for that.

I don't know if there is any licence to pay for that, it seems not. I found this answer and data about this protocol: 
https://stackoverflow.com/questions/69809003/is-there-a-standard-way-of-sending-gps-navigation-over-ble
The documentation is deep !

Thanks !

**Comment by rinigus** [comment]
This is, in principle, supported via DBus interface. Pure Maps DBus interface was used for smartwatch connection (https://github.com/piggz/harbour-amazfish/blob/master/daemon/src/navigationinterface.h). To get connection to other devices, you would need corresponding daemon that would export those signals. And that, in turn, will depend on each specific device. So, all you have to do is to write such device specific interface. Pure Maps side is ready for it already.

**Reply by Epy** [reply]
Thank you for your answer :)

Sorry I'm not sure I fully understand (and can't understand code, mostly) :

As I understand you already send navigation informations to a smartwatch, and so, already over bluetooth ?
If the protocol over bluetooth is really standardized, it should work for any device and not require any more developpement.

And by devices I think about some screen like: https://beeline.co/ or same thing integrated (in motorbikes for example) they seem to work with "any" GPS apps on your phone, at least proprietary common ones because they don't say to install their own smartphone app:
https://support.beeline.co/en/support/solutions/articles/77000227377-does-beeline-moto-work-with-my-phone-

So the navigation application on the phone sends data to the OS, which reads that and uses the protocol linked above for any device listening on bluetooth.
It could be the same with PureMaps but using DBus to send to a small program which will translate instructions to that protocol.

I almost could try that in Python (if I had a device for testing)

I'm not sure for all of that, it's assumptions.

Edit: And I just read that they DO have an navigation app to install, so it's a wrong example :disappointed: . I wonder if it's the same for the motorcycle I was watching, I'll ask.

**Reply by rinigus** [reply]
As far as I understand, it is NOT standardized and some kind of an app that translates to correct bluetooth protocol is needed. An example above is such app. But I never looked into it, that translation was done for smartwatch by Piggz.
